app/models/user.rb

```diff
@@ -81,28 +81,51 @@ class User < ApplicationRecord
   validates :locale, inclusion: {
     in: Rails.application.config.i18n.available_locales.map(&:to_s).push(nil)
   }
-  validate :password_complexity
+  validate :password_complexity, unless: :oauth_provided?
 
   def active_for_authentication?
     super && !banned
   end
 
-  def self.find_for_google_oauth2(access_token)
-    user = find_or_initialize_by(email: access_token.info.email)
-    user.name ||= access_token.info.name
+  def self.find_for_google_oauth2(auth)
+    user = find_or_initialize_by(email: auth.info.email)
+    user.name ||= auth.info.name
     user.password ||= Devise.friendly_token[0, 20]
-    update_access_token_fields(user: user, access_token: access_token)
+    update_access_token_fields(user: user, access_token: auth)
     user
   end
 
+  def self.find_for_facebook(auth)
```

Comment:
consider merging this method with the `def self.find_for_google_oauth2(auth)` method into
```suggestion
  def self.find_for_oauth(auth)
```
as hinted https://github.com/ifmeorg/ifme/pull/1705/files?file-filters%5B%5D=.rb#r411277154